### PR TITLE
RFR: Add bootstrap batch file. Adjust yarn commands

### LINKED ISSUE #204: conda activate command doesn't work in `yarn`
So the problem I was having was that, not matter how I try to slice it, I *cannot* get a `conda activate` command to run via yarn. EVen in it's own command, even using the various different syntax options.

My batch files *did* work, just saying... ;)

### PR DESCRIPTION
**Pull request by Centrus007**
Resolves #204

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -3,8 +3,7 @@
   "private": true,
   "scripts": {
     "bootstrap": "yarn && yarn conda:forge && yarn dependencies && yarn nbstripout:install && yarn install:dev",
```
Comment (Centrus007) [inline package.json]: could add a `yarn git:clean` here?

**Comment by Centrus007** [comment]
Btw, installation does appear to have worked ish, but when I try to commit, I get this:

```bash
λ git commit -m "Add bootstrap batch file.
husky > pre-commit (node v10.15.0)
'nbstripout' is not recognized as an internal or external command,
operable program or batch file.
husky > pre-commit hook failed (add --no-verify to bypass)
```

File: package.json (hunk 1 above)
Comment (SimonBiggs) [reply package.json]: Ooo, don't think so... hmmm... that one I think we need to leave up to people actively deciding to do it. `git clean` can be a little nuclear... 

If they have just cloned the repo, `git clean` won't be needed. If they have things left over in the repo and it is causing issues, I think it needs to be a choice to do it.

Maybe though we could have a bat script called "start fresh" that does the following:

```batch
call conda deactivate
yarn conda:delete
yarn git:clean
start ./bootstrap-with-conda-env.bat
```

File: bootstrap-with-conda-env.bat [hunk 2]
```diff
@@ -0,0 +1,3 @@
+yarn conda:create &&^
+conda activate pmp &&^
+yarn bootstrap
```
Comment (SimonBiggs) [inline bootstrap-with-conda-env.bat]: Awesome :)

File: bootstrap-with-conda-env.bat [hunk 3]
```diff
@@ -0,0 +1,3 @@
+yarn conda:create &&^
+conda activate pmp &&^
```
Comment (SimonBiggs) [inline bootstrap-with-conda-env.bat]: do you need `call conda activate pmp` here?

Also, what does the `&&^` mean?

(only if you think you need it...)
```suggestion
call conda activate pmp &&^
```

File: package.json [hunk 4]
```diff
@@ -3,8 +3,7 @@
...
-    "conda:activate": "conda activate pmp",
-    "conda:create": "conda create -y --name pmp python=3.7 && yarn conda:activate && yarn bootstrap",
+    "conda:create": "conda create -y --name pmp python=3.7",
     "conda:delete": "conda env remove -y --name pmp",
     "conda:forge": "conda config --add channels conda-forge",
     "conda:reinstall": "yarn conda:delete && yarn conda:create",
```
Comment (SimonBiggs) [inline package.json]: We should probably delete this line also, given it now no longer reinstalls...

```suggestion
```

**Comment by SimonBiggs** [comment]
> Btw, installation does appear to have worked ish, but when I try to commit, I get this:
> 
> ```shell
> λ git commit -m "Add bootstrap batch file."
> husky > pre-commit (node v10.15.0)
> 'nbstripout' is not recognized as an internal or external command,
> operable program or batch file.
> husky > pre-commit hook failed (add --no-verify to bypass)
> ```

Which I believe means `nbstripout` isn't find-able in your path. Does `nbstripout --status` also say `'nbstripout' is not recognized as an internal or external command, operable program or batch file.`?

File: bootstrap-with-conda-env.bat (hunk 3 above)
Comment (Centrus007) [reply bootstrap-with-conda-env.bat]: - `&&` only executes the next command if the previous command executed successfully (status code 0, I believe)
- `^` is a line continuation flag.
- As I understand it, `call` kinda achieves the opposite of `&&`, the next command will run regardless of the success of the current command...

**Comment by Centrus007** [comment]
> Which I believe means nbstripout isn't find-able in your path. Does nbstripout --status also say 'nbstripout' is not recognized as an internal or external command, operable program or batch file.

Nope.

**Comment by SimonBiggs** [comment]
I suspect there is something going on with nbstripout and your environment. Which environments is nbstripout installed in?

**Comment by SimonBiggs** [comment]
Is it an issue of `call conda activate pmp &&^` vs `conda activate pmp &&^`?

File: bootstrap-with-conda-env.bat (hunk 3 above)
Comment (SimonBiggs) [reply bootstrap-with-conda-env.bat]: I believe `call` says, run that `.bat` file then come back here with those changes made. I think it is needed for `conda activate` within a bat file...

It's so funny (frustrating) that `&&` actually needs to be written and isn't the default behavior.

`^` line continuation?

File: start-fresh.bat [hunk 5]
```diff
@@ -0,0 +1,4 @@
+call conda deactivate &&^
+yarn conda:delete &&^
+yarn git:clean &&^
+call ./bootstrap-with-conda-env.bat
```
Comment (SimonBiggs) [inline start-fresh.bat]: Just ran this here on the work PC, it worked, and so too did `nbstripout` hmm...

**Comment by Centrus007** [comment]
```bash
C:\Users\mjenni03\PythonProjects\pymedphys (extract-film-plane-from-DICOM -> origin)
(planqa) λ activate pmp

conda.bat activate pmp

C:\Users\mjenni03\PythonProjects\pymedphys (extract-film-plane-from-DICOM -> origin)
(pmp) λ conda list | grep nbstripout
WARNING: The conda.compat module is deprecated and will be removed in a future release.
nbstripout                0.3.5                    pypi_0    pypi

C:\Users\mjenni03\PythonProjects\pymedphys (extract-film-plane-from-DICOM -> origin)
(pmp) λ nbstripout --status
nbstripout is installed in repository b'C:\\Users\\mjenni03\\PythonProjects\\pymedphys'

Filter:
  clean = b'"c:/users/mjenni03/appdata/local/continuum/anaconda3/envs/pmp/python.exe" "c:/users/mjenni03/appdata/local/continuum/anaconda3/envs/pmp/lib/site-packages/nbstripout"'
  smudge = b'cat'
  required = b'true'
  diff= b'"c:/users/mjenni03/appdata/local/continuum/anaconda3/envs/pmp/python.exe" "c:/users/mjenni03/appdata/local/continuum/anaconda3/envs/pmp/lib/site-packages/nbstripout" -t'
  extrakeys=

Attributes:
  b'*.ipynb: filter: nbstripout'

Diff Attributes:
  b'*.ipynb: diff: ipynb'

C:\Users\mjenni03\PythonProjects\pymedphys (extract-film-plane-from-DICOM -> origin)
(pmp) λ touch test.py

C:\Users\mjenni03\PythonProjects\pymedphys (extract-film-plane-from-DICOM -> origin)
(pmp) λ git add test.py

C:\Users\mjenni03\PythonProjects\pymedphys (extract-film-plane-from-DICOM -> origin)
(pmp) λ git commit -m "add test.py"
husky > pre-commit (node v10.15.0)
[extract-film-plane-from-DICOM 7c8195f] add test.py
 1 file changed, 0 insertions(+), 0 deletions(-)
 create mode 100644 test.py
```
Maybe working now??

**Comment by SimonBiggs** [comment]
What did you change?

**Comment by Centrus007** [comment]
> What did you change?

Ummmm...........

Dunno.

**Comment by SimonBiggs** [comment]
I added `call` in ...? Might that have done it?

**Comment by SimonBiggs** [comment]
I'm happy for this to merge...

**Pull request merged**